src/algebra/group/units.lean

```diff
@@ -368,14 +386,6 @@ lemma is_unit.mul_iff [comm_monoid M] {x y : M} : is_unit (x * y) ↔ is_unit x
 ⟨λ h, ⟨is_unit_of_mul_is_unit_left h, is_unit_of_mul_is_unit_right h⟩,
   λ h, is_unit.mul h.1 h.2⟩
 
-@[to_additive] theorem is_unit.mul_right_inj [monoid M] {a b c : M} (ha : is_unit a) :
-  a * b = a * c ↔ b = c :=
-by cases ha with a ha; rw [←ha, units.mul_right_inj]
-
-@[to_additive] theorem is_unit.mul_left_inj [monoid M] {a b c : M} (ha : is_unit a) :
-  b * a = c * a ↔ b = c :=
-by cases ha with a ha; rw [←ha, units.mul_left_inj]
```

Comment:
Note the old proof did not use choice; I don't think there's any reason for this PR to introduce it.

Reply:
If you really care about character count, then you can use
```lean
let ⟨u, hu⟩ := ha in hu.symm ▸ u.mul_left_inj
```
or similar. Avoiding choice is more important than golfing one trivial proof into another one that uses fewer characters IMO.

Reply:
It was less about characters than about showing how the API flows, but I okay I get it.